src/Mod/TechDraw/App/AppTechDrawPy.cpp

```diff
@@ -961,18 +961,18 @@ class Module : public Py::ExtensionModule<Module>
     {
         PyObject* pFace(nullptr);
         double scale = 1.0;
-        char* pPatName = "";
-        char* pPatFile = "";
+        char* pPatName = nullptr;

```

Comment:
Why not using `const char* pPatName = "";`? The you can also avoid the ternary operator below.

Reply:
Yes, good idea. I have changed it.

Reply:
On second thought, I'm not confident in my ability to make this change without changing the behavior -- right now it's using an "et" input type in `PyArg_ParseTuple`, which allocates a buffer (that needs to be freed). How can I change this call:
```
        if (!PyArg_ParseTuple(args.ptr(), "O|detet", &pFace, &scale, "utf-8", &pPatName, "utf-8", &pPatFile)) {
```
to use a `const char *` for `pPatName` and `pPatFile` without changing the encoding behavior?

Reply:
@wwmayer do you have a minute to help me out here? I don't have a good understanding of why that `et` type is being used.

Reply:
#11014 Core: Enable compiling with MSVC /permissive-
https://github.com/WandererFan/FreeCAD/tree/PR11014
[TD]Use regular py string not encoded string in PyArg_ParseTuple

Works with áéöĀ in the file name and the pattern name for Draft Hatch.

Reply:
Thanks, I've applied that change.